compiler/src/dmd/parse.d

```diff
@@ -9515,14 +9515,9 @@ LagainStc:
 
     void usageOfBodyKeyword()
     {
-        version (none)
-        {
-            // @@@DEPRECATED_2.117@@@
-            // https://github.com/dlang/DIPs/blob/1f5959abe482b1f9094f6484a7d0a3ade77fc2fc/DIPs/accepted/DIP1003.md
-            // Deprecated in 2.097 - Can be removed from 2.117
-            // The deprecation period is longer than usual as `body`
-            // was quite widely used.
-            deprecation("usage of the `body` keyword is deprecated. Use `do` instead.");
+        if (compileEnv.obsolete)
+        {
+            eSink.warning(token.loc, "usage of identifer `body` as a keyword is obsolete. Use `do` instead.");
```

Comment:
I'm currently swaying towards rethinking how we do warnings/errors.  Possibly generating an enum from all the field members of `Param`, then adding overloads of these functions that have said enum as the first argument.

Think.
```
eSink.warning(GlobalParams.obsolete, token.loc, "usage of identifier `body` as a keyword is obsolete...");
```
It'll then be on the warning routine to map the enum value `GlobalParams.obsolete` back to its field, and check whether to emit the diagnostic or not.

Reply:
The idea was to have eSink be an abstract interface, and have multiple implementations of it. This enabled lexer/parse/cparse to be independent of the baroque and overly complex implementation in errors.d. This was successful, and even allowed "mock" implementations for unittests that avoided problems like error gagging.